urbanpy/download/download.py

```diff
@@ -513,3 +524,218 @@ def hdx_dataset(resource):
+        Year for satellite data (2017-2024 available)
+    bands : list, optional
+        Embedding bands to download (A00-A63). If None, uses all 64 bands.
+    reducer : str, default 'mean'
+        Reduction method for pixels within each geometry: 'mean', 'median', 'first'
+    scale : int, default 10
+        Pixel resolution in meters (native is 10m)
+    tile_scale : int, default 1
+        Tile scale factor for memory management (increase for large areas)
+    
+    Returns
+    -------
+    GeoDataFrame
+        Input geometries with embedding columns (A00, A01, ...) added
+        
+    Examples
+    --------
+    >>> # Download city and create hexagons
+    >>> lima = up.download.nominatim_osm("Lima, Peru", email="user@email.com")
+    >>> hexes = up.geom.gen_hexagons(8, lima)
+    >>> 
+    >>> # Download all 64 embedding bands
+    >>> embeddings = up.download.google_satellite_embeddings(hexes, year=2023)
+    >>> 
+    >>> # Use with different reducer methods
+    >>> embeddings_median = up.download.google_satellite_embeddings(
+    ...     hexes, year=2023, reducer='median'
+    ... )
+    
+    Notes
+    -----
+    Requires Earth Engine authentication. Run `ee.Authenticate()` and `ee.Initialize()`
+    before using this function.
+    
+    The embeddings are "linearly composable", meaning they can be aggregated 
+    while retaining semantic meaning. Use `up.geom.resolution_downsampling()` 
+    to aggregate fine H3 hexagons to coarser resolutions.
+    """
+    
+    if not EE_AVAILABLE:
+        raise ImportError(
+            "Earth Engine not available. Install with: pip install earthengine-api"
+        )
+
+    if len(gdf) > 5000:
+        raise ValueError(f"Too many geometries ({len(gdf)}). Maximum 5,000 supported. Try batch processing.")
+    
+    if year < 2017 or year > 2024:
+        raise ValueError(f"Year {year} not supported. Available years: 2017-2024")
+    
+    if reducer not in ['mean', 'median', 'first']:
+        raise ValueError(f"Unsupported reducer: {reducer}. Use 'mean', 'median', or 'first'")
+    
+    if scale < 1:
+        raise ValueError(f"Scale must be positive, got {scale}")
+    
+    # Use all 64 bands by default
+    if bands is None:
+        bands = [f'A{i:02d}' for i in range(64)]
+    
+    # Validate bands
+    valid_bands = [f'A{i:02d}' for i in range(64)]
+    invalid_bands = [b for b in bands if b not in valid_bands]
+    if invalid_bands:
+        raise ValueError(f"Invalid bands: {invalid_bands}. Valid bands are A00-A63")
+    
+    # Load collection and filter
+    collection = ee.ImageCollection('GOOGLE/SATELLITE_EMBEDDING/V1/ANNUAL')
+    start_date = f"{year}-01-01"
+    end_date = f"{year}-12-31"
+    
+    # Filter by date and bounds
+    bounds = gdf.total_bounds
+    roi = ee.Geometry.Rectangle([bounds[0], bounds[1], bounds[2], bounds[3]])
+    
+    filtered = collection.filterDate(start_date, end_date).filterBounds(roi)
+    
+    if filtered.size().getInfo() == 0:
```

Comment:
According to the docs, `.size()` should return an integer with the number of features. Could we remove `.getInfo()`? Should we do `.getInfo()` after we know the size is bigger than 0?

`if filtered.size() == 0`


Docs:
- [size](https://developers.google.com/earth-engine/apidocs/ee-imagecollection-size)
- [getInfo](https://developers.google.com/earth-engine/apidocs/ee-imagecollection-getinfo)